src/pe/resource.rs

```diff
@@ -0,0 +1,1681 @@
+        }
+
+        debug_struct
+            .field(
+                "value_slice",
+                &format_args!(
+                    "{:02x?} ({} bytes, {})",
+                    self.value,
+                    self.value.len(),
+                    if self.value.len() == self.value_len as usize {
+                        "Correct"
+                    } else {
+                        "Incorrect"
+                    }
+                ),
+            )
+            .finish()
+    }
+}
+
+impl<'a> ResourceString<'a> {
+    pub fn parse(bytes: &'a [u8], offset: &mut usize) -> error::Result<Self> {
+        let len = bytes.gread_with::<u16>(offset, scroll::LE)?;
+        let value_len = bytes.gread_with::<u16>(offset, scroll::LE)?;
+        let r#type = bytes.gread_with::<u16>(offset, scroll::LE)?;
+        *offset = align_up(*offset, RESOURCE_STRING_FIELD_ALIGNMENT);
+        let key_size = &bytes[*offset..]
+            .chunks(2)
+            .take_while(|x| u16::from_le_bytes([x[0], x[1]]) != 0u16)
+            .count()
+            * SIZE_OF_WCHAR;
+        if (*offset - SIZE_OF_WCHAR) + key_size + SIZE_OF_WCHAR > bytes.len() {
+            return Err(error::Error::Malformed(format!(
+                "offset ({:#x}) and key_size ({:#x}) is greater than bytes len {:#x}",
+                offset,
+                key_size,
+                bytes.len()
+            )));
+        }
+        let key =
+            &bytes[*offset - SIZE_OF_WCHAR..*offset - SIZE_OF_WCHAR + key_size + SIZE_OF_WCHAR];
+        *offset += align_up(key.len(), RESOURCE_STRING_FIELD_ALIGNMENT);
+        let real_value_len = align_up(
+            if r#type == 1 {
+                value_len as usize * SIZE_OF_WCHAR
+            } else {
+                value_len as usize
+            },
+            4,
+        );
+        if *offset + (real_value_len as usize) > bytes.len() {
+            return Err(error::Error::Malformed(format!(
+                "offset ({:#x}) and real_value_len ({:#x}) is greater than bytes len {:#x}",
+                offset,
+                real_value_len,
+                bytes.len()
+            )));
+        }
+        let value = &bytes[*offset..*offset + real_value_len as usize];
+        *offset += value.len();
+        Ok(Self {
+            len,
+            value_len,
+            r#type,
+            key,
+            value,
+        })
+    }
+
+    /// Returns `true` if [`ResourceString::value`] is expected to an null-terminated unicode string
+    pub fn is_text_data(&self) -> bool {
+        self.r#type == 1
+    }
+
+    /// Returns `true` if [`ResourceString::value`] is expected to a binary data
+    pub fn is_binary_data(&self) -> bool {
+        self.r#type == 0
+    }
+
+    /// Converts [`ResourceString::key`] into a [`String`]
+    pub fn key_string(&self) -> String {
+        to_utf16_string(&self.key)
+    }
+
+    /// Converts [`ResourceString::value`] into a [`String`]
+    pub fn value_string(&self) -> String {
```

Comment:
should this be `-> Option<String>`, since if `is_binary_data` is true, a string representation doesn't make sense?